pkg/cache/snapshot.go

```diff
@@ -66,7 +66,7 @@ func (s *Snapshot) Log(log logr.Logger) {
 			"cohort", cohortName,
 			"resourceGroups", cq.ResourceGroups,
 			"usage", cq.ResourceNode.Usage,
-			"workloads", utilmaps.Keys(cq.Workloads),
+			"workloads", maps.Keys(cq.Workloads),
```

Comment:
```suggestion
			"workloads", slices.Collect(maps.Keys(cq.Workloads)),
```

I don't think this will work well. Please collect it here as well.

Reply:
sure